lib/shared/require-or-import.js

```diff
@@ -0,0 +1,25 @@
+'use strict';
+
+var importESM;
+try {
+  importESM = new Function('id', 'return import(id);');
+} catch (e) {
+  importESM = null;
+}
```

Comment:
Node.js <10 errors out with a SyntaxError when loading a script that uses `import()`. So I dynamically create a function for that and catch the SyntaxError. That way we can keep supporting all Node.js versions all the way back to 0.10.